packages/blockchain-wallet-v4-frontend/src/data/components/interest/selectors.ts

```diff
@@ -128,12 +139,52 @@ export const getInstrumentsSortedByBalance = createDeepEqualSelector(
         return [...preferredRewardsCoins, ...rewardsInstruments]
       }
 
-      return [
+      const sortedInstruments = [
         ...preferredStakingCoins,
         ...stakingInstruments,
         ...preferredRewardsCoins,
         ...rewardsInstruments
-      ]
+      ].filter(({ coin, product }) => {
+        switch (earnTab) {
+          case 'All':
+            break
+          case 'Rewards':
+            if (product !== earnTab) return false
+            break
+          case 'Staking':
+            if (product !== earnTab) return false
+            break
+
+          default:
+            break
+        }
+
+        if (showAvailableAssets) {
+          const isStaking = product === 'Staking'
+          const account = isStaking
+            ? stakingBalances && stakingBalances[coin]
+            : rewardsBalances && rewardsBalances[coin]
+          const accountBalanceBase = account ? account.balance : 0
+          const hasAccountBalance = accountBalanceBase > 0
+
+          if (!hasAccountBalance) return false
+        }
+
+        const isCoinInWindow = !!window.coins[coin]
+
+        if (isCoinInWindow) {
+          const { displaySymbol, name, symbol } = window.coins[coin].coinfig
```

Comment:
`window.coins`

Let's try to avoid referencing `window` directly. There is `getCoins` selector in `packages/blockchain-wallet-v4/src/redux/data/coins/selectors.ts`.